doc/releases/changelog-dev.md

```diff
@@ -63,31 +84,26 @@
 * Updates to some relevant Pytests to enable its use as a suite of benchmarks.
   [(#4703)](https://github.com/PennyLaneAI/pennylane/pull/4703)
 
-* The function ``qml.equal`` now supports ``ControlledSequence`` operators.
-  [(#4829)](https://github.com/PennyLaneAI/pennylane/pull/4829)
-
 * Added `__iadd__` method to PauliSentence, which enables inplace-addition using `+=`, we no longer need to perform a copy, leading to performance improvements.
   [(#4662)](https://github.com/PennyLaneAI/pennylane/pull/4662) 
 
-* `qml.ArbitraryUnitary` now supports batching.
-  [(#4745)](https://github.com/PennyLaneAI/pennylane/pull/4745)
+<h4>Other improvements</h4>
 
-* `qml.draw` and `qml.draw_mpl` now render operator ids.
-  [(#4749)](https://github.com/PennyLaneAI/pennylane/pull/4749)
+* Added `ops.functions.assert_valid` for checking if an `Operator` class is defined correctly.
+  [(#4764)](https://github.com/PennyLaneAI/pennylane/pull/4764)
+
+* `GlobalPhase` now decomposes to nothing, in case devices do not support global phases.
+  [(#4855)](https://github.com/PennyLaneAI/pennylane/pull/4855)
 
-* `has_matrix` will now automatically be `True` if `Operator.matrix` is overridden, even if
+* Custom operations can now provide their matrix directly through the `Operator.matrix()` method
+  without requiring `Operator.compute_matrix()` to be provided. Behind the scenes,
+  `has_matrix` will now automatically be `True` if `Operator.matrix` is overridden, even if
```

Comment:
This was updated a bit more to emphasize the user-facing consequence.

Reply:
I'm ok with the change, but we never really required `Operator.compute_matrix` to be overridden. You just then had to manually update `has_matrix` to get device support.

Reply:
```suggestion
* Custom operations can now provide their matrix directly through the `Operator.matrix()` method
  without needing to update the `has_matrix` property. `has_matrix` will now automatically be
  `True` if `Operator.matrix` is overridden, even if
```
@albi3ro does this look better?